### PR TITLE
[improve][broker] Make some operation messageTTL methods in Namespaces async.

### PR DESCRIPTION
**Pull request by shibd**
### Motivation
See PIP https://github.com/apache/pulsar/issues/14365 and change tracker https://github.com/apache/pulsar/issues/15043.

Make NamespacesBase `getNamespaceMessageTTL / setNamespaceMessageTTL / removeNamespaceMessageTTL ` methods to pure async.

### Documentation
- [x]  no-need-doc

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 1]
```diff
@@ -405,10 +411,16 @@ public Integer getNamespaceMessageTTL(@PathParam("property") String property, @P
...
-            @PathParam("namespace") String namespace, int messageTTL) {
+    public void setNamespaceMessageTTL(@Suspended AsyncResponse asyncResponse, @PathParam("property") String property,
+                                       @PathParam("cluster") String cluster, @PathParam("namespace") String namespace,
+                                       int messageTTL) {
         validateNamespaceName(property, cluster, namespace);
-        internalSetNamespaceMessageTTL(messageTTL);
+        internalSetNamespaceMessageTTLAsync(messageTTL)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
+                .exceptionally(ex -> {
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:421]: I think it's better to add log
 'log.error("Failed to set namespace message TTL for namespace {}", namespaceName, ex);'

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 2]
```diff
@@ -390,13 +390,19 @@ public void setNamespaceReplicationClusters(@PathParam("property") String proper
...
-        validateNamespacePolicyOperation(NamespaceName.get(property, namespace), PolicyName.TTL, PolicyOperation.READ);
-
-        Policies policies = getNamespacePolicies(namespaceName);
-        return policies.message_ttl_in_seconds;
+        validateNamespacePolicyOperationAsync(NamespaceName.get(property, namespace), PolicyName.TTL,
+                PolicyOperation.READ)
+                .thenCompose(__ -> getNamespacePoliciesAsync(namespaceName))
+                .thenAccept(policies -> asyncResponse.resume(policies.message_ttl_in_seconds))
+                .exceptionally(ex -> {
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:403]: Add error.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java [hunk 3]
```diff
@@ -417,10 +429,17 @@ public void setNamespaceMessageTTL(@PathParam("property") String property, @Path
...
+    public void removeNamespaceMessageTTL(@Suspended AsyncResponse asyncResponse,
+                                          @PathParam("property") String property,
+                                          @PathParam("cluster") String cluster,
+                                          @PathParam("namespace") String namespace) {
         validateNamespaceName(property, cluster, namespace);
-        internalSetNamespaceMessageTTL(null);
+        internalSetNamespaceMessageTTLAsync(null)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
+                .exceptionally(ex -> {
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/Namespaces.java:440]: add log

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 4]
```diff
@@ -333,13 +333,17 @@ public void setNamespaceReplicationClusters(@PathParam("tenant") String tenant,
...
-        validateNamespacePolicyOperation(NamespaceName.get(tenant, namespace), PolicyName.TTL, PolicyOperation.READ);
-
-        Policies policies = getNamespacePolicies(namespaceName);
-        return policies.message_ttl_in_seconds;
+        validateNamespacePolicyOperationAsync(NamespaceName.get(tenant, namespace), PolicyName.TTL,
+                PolicyOperation.READ)
+                .thenCompose(__ -> getNamespacePoliciesAsync(namespaceName))
+                .thenAccept(policies -> asyncResponse.resume(policies.message_ttl_in_seconds))
+                .exceptionally(ex -> {
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:344]: add log

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 5]
```diff
@@ -348,10 +352,17 @@ public Integer getNamespaceMessageTTL(@PathParam("tenant") String tenant,
...
+    public void setNamespaceMessageTTL(@Suspended AsyncResponse asyncResponse, @PathParam("tenant") String tenant,
+                                       @PathParam("namespace") String namespace,
+                                       @ApiParam(value = "TTL in seconds for the specified namespace", required = true)
+                                               int messageTTL) {
         validateNamespaceName(tenant, namespace);
-        internalSetNamespaceMessageTTL(messageTTL);
+        internalSetNamespaceMessageTTLAsync(messageTTL)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
+                .exceptionally(ex -> {
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:363]: add log

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java [hunk 6]
```diff
@@ -360,10 +371,16 @@ public void setNamespaceMessageTTL(@PathParam("tenant") String tenant, @PathPara
...
-    public void removeNamespaceMessageTTL(@PathParam("tenant") String tenant,
+    public void removeNamespaceMessageTTL(@Suspended AsyncResponse asyncResponse,
+                                          @PathParam("tenant") String tenant,
                                           @PathParam("namespace") String namespace) {
         validateNamespaceName(tenant, namespace);
-        internalSetNamespaceMessageTTL(null);
+        internalSetNamespaceMessageTTLAsync(null)
+                .thenAccept(__ -> asyncResponse.resume(Response.ok().build()))
+                .exceptionally(ex -> {
+                    resumeAsyncResponseExceptionally(asyncResponse, ex);
```
Comment (Technoboy-) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/Namespaces.java:381]: And here

**Review by Technoboy- — APPROVED** [review]
LGTM

**Comment by shibd** [comment]
@Technoboy-  Thanks for your review, all log added.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**